### PR TITLE
feat: add 20-bullseye-slim image

### PR DESCRIPTION
**Pull request by jltimm**
Adds Node 20 bullseye image.

## New Image Checklist

If you're adding a new image, make sure you have done the following.

* [x] Added to lint workflow matrix (`.github/workflows/lint.yml`)
* [x] Added to build workflow matrix (`.github/workflows/build.yml`)
* [x] Added to Makefile (`Makefile`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,10 +10,11 @@ FROM articulate/articulate-node:<tag>
 
 > 🌟 recommended image
 
+* __20-bullseye-slim__ 🌟
 * __18-bullseye-slim__ 🌟
 * __18-lambda__ 🌟
-* __16-bullseye-slim__ 🌟
-* __16-lambda__ 🌟
+* 16-bullseye-slim
+* 16-lambda
```
Comment (jltimm) [inline README.md:17]: node 16 end of life [is 9/11/2023](https://nodejs.org/en/blog/announcements/nodejs16-eol)

**Comment by mloberg** [comment]
@jltimm How urgent do you need this? We are talking about reworking our base images and we'd want to do that starting with Node 20

**Comment by jltimm** [comment]
@mloberg not very urgent. We wanted to upgrade our teams node images to 20 in order to give us more time until the next upgrade, but we can always upgrade to 18 in the meantime.

Do you have any timeline on when you'd be finished reworking the base images? If not obviously that's 100% fine

**Comment by mloberg** [comment]
I added a new image Node 20 in #106. This follows our new base image spec, so it might not be as easy as changing the base tag. `articulate/node:20`

**Pull request closed without merging**